### PR TITLE
Remove usage of deprecated `data` field in Neutron SDK ICTX helper for SDK 0.47 chains & update SDK 0.45 helper to backw compat NTRN-223

### PR DESCRIPTION
**Pull request by quasisamurai**
[TASK](https://hadronlabs.atlassian.net/browse/NTRN-223)

related prs:
https://github.com/neutron-org/neutron-dev-contracts/pull/43
https://github.com/neutron-org/neutron-integration-tests/pull/271

### CODE REVIEW

File: packages/neutron-sdk/src/interchain_txs/v047/helpers.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const ICA_OWNER_DELIMITER: &str = ".";
+
+/// Constructs a full ICA controller port identifier for a contract with **contract_address** and **interchain_account_id**
+/// <https://github.com/cosmos/ibc-go/blob/46e020640e66f9043c14c53a4d215a5b457d6703/modules/apps/27-interchain-accounts/types/port.go#L11>
+pub fn get_port_id<R: AsRef<str>>(contract_address: R, interchain_account_id: R) -> String {
+    CONTROLLER_PORT_PREFIX.to_string()
+        + contract_address.as_ref()
+        + ICA_OWNER_DELIMITER
+        + interchain_account_id.as_ref()
+}
```
Comment (pr0n00gler) [inline packages/neutron-sdk/src/interchain_txs/v047/helpers.rs:36]: I think it's better to have common helpers in the root file, not in a package

Comment (pr0n00gler) [reply packages/neutron-sdk/src/interchain_txs/v047/helpers.rs:36]: Please remove this helpers from v047 package as we discussed

File: packages/neutron-sdk/src/interchain_txs/v045/helpers.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                        })
+                    })
+                    .collect::<StdResult<Vec<MsgData>>>()
+            } else {
+                // Field `.data` is deprecated since cosmos-sdk v047.
+                // But for backwards compatibility we still allow that. Given function can be used w both v045 & v047
+                #[allow(deprecated)]
+                Ok(msg.data)
+            }
+        }
```
Comment (sotnikov-s) [inline packages/neutron-sdk/src/interchain_txs/v045/helpers.rs:34]: we decided to remove the branch for v047 from here, didn't we?

File: packages/neutron-sdk/src/interchain_txs/v045/helpers.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+use cosmos_sdk_proto::{
+    cosmos::base::abci::v1beta1::{MsgData, TxMsgData},
+    traits::Message,
+};
+use cosmwasm_std::{Binary, StdError, StdResult};
+
+/// Decodes acknowledgement into `Vec<MsgData>` structure
+/// We consider this method as deprecated. Use v047 instead.
```
Comment (sotnikov-s) [inline packages/neutron-sdk/src/interchain_txs/v045/helpers.rs:8]: do we still need this comment line?

**Review by sotnikov-s — COMMENTED** [review]
w8ing for fixes re

**Review by pr0n00gler — APPROVED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Pull request merged**